Chatto/sources/ChatController/ChatMessages/ChatMessageCollectionAdapter.swift

```diff
@@ -0,0 +1,702 @@
+//
+// Copyright (c) Badoo Trading Limited, 2010-present. All rights reserved.
+//
+
+import UIKit
+
+public protocol ChatMessageCollectionAdapterProtocol: UICollectionViewDataSource, UICollectionViewDelegate {
+    var chatItemCompanionCollection: ChatItemCompanionCollection { get }
+    var delegate: ChatMessageCollectionAdapterDelegate? { get set }
+
+    func startProcessingUpdates()
+    func stopProcessingUpdates()
+    func setup(in collectionView: UICollectionView)
+
+    // TODO: Remove from the adapter
+    func indexPath(of itemId: String) -> IndexPath?
+    func refreshContent(completionBlock: (() -> Void)?)
+    func scheduleSpotlight(for: String)
+}
+
+public protocol ChatMessageCollectionAdapterDelegate: AnyObject {
+    var isFirstLoad: Bool { get }
+
+    func chatMessageCollectionAdapter(_ : ChatMessageCollectionAdapterProtocol, onDisplayCellWithIndexPath: IndexPath)
+    func chatMessageCollectionAdapter(_ : ChatMessageCollectionAdapter, didUpdateItemsWithUpdateType: UpdateType)
+    func chatMessageCollectionAdapterShouldAnimateCellOnDisplay(_ : ChatMessageCollectionAdapterProtocol) -> Bool
```

Comment:
I feel like it's much less readable to pass self in the delegate methods, and exposes unnecessary dependencies to callee.

I can see why Apple does this for their UIKit delegate protocols, but not sure if it provides any value for application code.

What do you think about removing the first parameter from Delegate protocols?